db/document.go

```diff
@@ -515,6 +515,9 @@ func parseXattrStreamData(xattrName string, userXattrName string, data []byte) (
 	}
 
 	xattrsLen := binary.BigEndian.Uint32(data[0:4])
+	if int(xattrsLen) > len(data) {
```

Comment:
This looks like it should check if int(xattrsLen+4) > len(data) to avoid all possible panics on line 518.

Reply:
Fixed.